math/mathcore/inc/Math/IFunction.h

```diff
@@ -360,7 +360,7 @@ namespace ROOT {
             Gradient(x, df);
          }
 
-
+         virtual bool returnsInMinuit2ParameterSpace() const { return false; }
```

Comment:
Is this function really needed ? It should not be at the gradient calculator level to know if the internal parameter space is used

Reply:
Looking at the code, I would add maybe a new flag to the 
`ROOT::Math::Minimiser::SetFunction( gradFunc & f, bool= false)`
In this way the Minuit2Minimizer can set this flag in the FCNGradientBase class when  creating it from the function interface passed by the clients (e.g. RooMinimizer).
If you prefer, we can merge with this and I can do this change later, please let me know

Reply:
Your suggestion makes sense, indeed it feels weird that a Minuit detail is in mathcore. In light of time constraints, I would prefer to keep it like this for now, if you don't mind too much. The problem is that returnsInMinuit2ParameterSpace is also used in other places right now, in particular in the `TestStatistics::MinuitFcnGrad` class that will be introduced in the next PR I will make.